src/routes/new-load/Load.tsx

```diff
@@ -0,0 +1,138 @@
+  loadSafeAddressStepValidations,
+} from './steps/LoadSafeAddressStep'
+import LoadSafeOwnersStep, { loadSafeOwnersStepLabel } from './steps/LoadSafeOwnersStep'
+import ReviewLoadStep, { reviewLoadStepLabel } from './steps/ReviewLoadStep'
+import { getRandomName } from 'src/logic/hooks/useMnemonicName'
+import StepperForm, { StepFormElement } from 'src/components/StepperForm/StepperForm'
+import SelectNetworkStep, { selectNetworkStepLabel } from './steps/SelectNetworkStep'
+import { isValidAddress } from 'src/utils/isValidAddress'
+import { makeAddressBookEntry } from 'src/logic/addressBook/model/addressBook'
+import { addressBookSafeLoad } from 'src/logic/addressBook/store/actions'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import { buildSafe } from 'src/logic/safe/store/actions/fetchSafe'
+import { loadStoredSafes, saveSafes } from 'src/logic/safe/utils'
+import { addOrUpdateSafe } from 'src/logic/safe/store/actions/addOrUpdateSafe'
+import { SAFE_ROUTES } from '../routes'
+import {
+  FIELD_LOAD_CUSTOM_SAFE_NAME,
+  FIELD_LOAD_IS_LOADING_SAFE_ADDRESS,
+  FIELD_LOAD_SAFE_ADDRESS,
+  FIELD_LOAD_SUGGESTED_SAFE_NAME,
+  FIELD_SAFE_OWNER_LIST,
+} from './fields/loadFields'
+
+function Load(): ReactElement {
+  const provider = useSelector(providerNameSelector)
+
+  const dispatch = useDispatch()
+
+  const classes = useStyles()
+
+  const { safeAddress } = useParams<{ safeAddress?: string }>()
+
+  const initialValues = {
+    [FIELD_LOAD_SUGGESTED_SAFE_NAME]: getRandomName('safe'),
+    [FIELD_LOAD_SAFE_ADDRESS]: safeAddress,
+    [FIELD_LOAD_IS_LOADING_SAFE_ADDRESS]: false,
+    [FIELD_SAFE_OWNER_LIST]: [],
+  }
+
+  const onSubmitLoadSafe = async (values) => {
+    const safeName = values[FIELD_LOAD_CUSTOM_SAFE_NAME] || values[FIELD_LOAD_SUGGESTED_SAFE_NAME]
+    const safeAddress = values[FIELD_LOAD_SAFE_ADDRESS]
+    const ownerList = values[FIELD_SAFE_OWNER_LIST]
+
+    if (!isValidAddress(safeAddress)) {
+      return
+    }
+
+    const ownerListWithNames = ownerList
+      .map((owner) => {
+        const ownerFieldName = `owner-address-${owner.address}`
+        const ownerNameValue = values[ownerFieldName]
+        return {
+          ...owner,
+          name: ownerNameValue,
+        }
+      })
+      .filter((owner) => !!owner.name)
+
+    const safeAddressBook = makeAddressBookEntry({
+      address: safeAddress,
+      name: safeName,
+    })
+    dispatch(addressBookSafeLoad([...ownerListWithNames, safeAddressBook]))
+
+    const checksumSafeAddress = checksumAddress(safeAddress)
+    const safeProps = await buildSafe(checksumSafeAddress)
+    const storedSafes = (await loadStoredSafes()) || {}
+    storedSafes[checksumSafeAddress] = safeProps
+    await saveSafes(storedSafes)
+    await dispatch(addOrUpdateSafe(safeProps))
+    history.push(
+      generatePath(SAFE_ROUTES.ASSETS_BALANCES, {
+        safeAddress,
+      }),
+    )
+  }
+
+  const isProductionEnv = process.env.REACT_APP_ENV === 'production'
```

Comment:
`process.env.REACT_APP_ENV` can be replaced with `APP_ENV` from `constants.ts`

Reply:
done!